pkg/graph/resolvers/it_gov_task_statuses_test.go

```diff
@@ -646,14 +653,215 @@ func TestBizCaseDraftStatus(t *testing.T) {
+				GRTDate: nil,
+			},
+			expectedStatus: models.ITGGRTSNotNeeded,
+			expectError:    false,
+		},
+		{
+			testCase: "Final Business Case Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepFINALBIZCASE,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "Final Business Case Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepFINALBIZCASE,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "GRB Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRBMEETING,
+				GRTDate: nil,
+			},
+			expectedStatus: models.ITGGRTSNotNeeded,
+			expectError:    false,
+		},
+		{
+			testCase: "GRB Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRBMEETING,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "GRB Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRBMEETING,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepDECISION,
+				GRTDate: nil,
+			},
+			expectedStatus: models.ITGGRTSNotNeeded,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepDECISION,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepDECISION,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    defaultTestStep,
+				GRTDate: nil,
+			},
+			expectedStatus: "",
+			expectError:    true,
+		},
+		{
+			testCase: "Decision Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    defaultTestStep,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    defaultTestStep,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
 	}
 
-	status := GrtMeetingStatus(&intake)
+	for _, test := range decisionStateTests {
+		t.Run(test.testCase, func(t *testing.T) {
+			status, err := GrtMeetingStatus(&test.intake)
+			assert.EqualValues(t, test.expectedStatus, status)
```

Comment:
This check should probably be moved under the `else` block for `if test.expectError` - we don't care about the returned status if an error is returned, so there's no point in having the test check it.

Reply:
I think that it is valid to check what the function is returning in this case. (since the function is returning a result, we expect it to be ""). This checks if the method suddenly returned something else and a status.

Reply:
That's a good point, we _should_ expect it to return `"", <some error>` in an error case, not any other value.